### PR TITLE
Adds ability to serve mocked data for MR API calls via cli flag

### PR DESCRIPTION
**Pull request by alexcreasy**
## Description

This adds scaffolding to be able to serve mocked data for MR API endpoints. It also incudes endpoints to mock all existing BFF endpoints and their tests. The remaining endpoints will be added shortly after in a separate PR

## How Has This Been Tested?

1. Run the backend with the `MOCK_MR_API=true` flag (See README updates for more info)
2. Run each of the example curl requests listed in the README and verify the output

## Merge criteria:

- All the commits have been [_signed-off_](https://github.com/kubeflow/community/tree/master/dco-signoff-hook#signing-off-commits)  (To pass the `DCO` check)

- [x] The commits have meaningful messages; the author will squash them after approval or in case of manual merges will ask to merge with squash.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work.
- [x] Code changes follow the [kubeflow contribution guidelines](https://www.kubeflow.org/docs/about/contributing/).

If you have UI changes

- [ ] The developer has added tests or explained why testing cannot be added.
- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Verify that UI/UX changes conform the UX guidelines for Kubeflow.

### CODE REVIEW

File: clients/ui/bff/README.md [hunk 1]
```diff
@@ -31,9 +31,9 @@ After building it, you can run our app with:
 ```shell
 make run
 ```
-If you want to use a different port or mock kubernetes client, useful for front-end development, you can run:
+If you want to use a different port, mock kubernetes client or model registry client - useful for front-end development, you can run:
 ```shell
-make run PORT=8000 MOCK_K8S_CLIENT=true
+make run PORT=8000 MOCK_K8S_CLIENT=true MOCK_MR_CLIENT=true
```
Comment (lucferbux) [inline clients/ui/bff/README.md:36]: Not a review, it's more of a question, are we gonna get 8000 as the default port? We have 4000 in the frontend for the api, if so, I'll do a follow up PR to change the references there.

Comment (ederign) [reply clients/ui/bff/README.md:36]: You are right, the default port seems to be PORT ?= 4000 (on Makefile)

Comment (ederign) [reply clients/ui/bff/README.md:36]: But on those instructions, I just to show how to do with an alternative one.

File: clients/ui/bff/api/helpers.go [hunk 2]
```diff
@@ -11,6 +11,8 @@ import (
 
 type Envelope map[string]interface{}
 
+type TypedEnvelope[T any] map[string]T
```
Comment (ederign) [inline clients/ui/bff/api/helpers.go:14]: @alexcreasy in a next iteration shall we remove the untyped one?

**Review by ederign — APPROVED** [review]

File: clients/ui/bff/api/helpers.go (hunk 2 above)
Comment (alexcreasy) [reply clients/ui/bff/api/helpers.go:14]: @ederign yes, I'd like to, the only reason I didn't just change it was for speed right now :)

**Pull request merged**